[Playbooks] - Playbook Section added (#900)
RevenueCat/docs: docs/playbooks/guides/freemium.mdx

sidebar_label: Freemium Paywalls
slug: freemium
---

This guide walks through implementing and optimizing a Freemium strategy using RevenueCat, focusing on how to manage entitlements, display upgrade prompts, and convert free users into paying customers with minimal friction.

## When to use

A freemium strategy provides value to users before asking them to pay. This model works well when:

- Offer a clear and compelling upgrade path from free to paid (e.g., feature gating, usage limits)
- Have low marginal costs per user (minimal server or infrastructure overhead)
- Leverage freemium users as growth drivers through sharing, referrals, or collaboration
- Face competitive pressure where users expect to try the core experience before subscribing

## Before you start

### 1. Define free features

- Core functionality
- Basic features that showcase value
- Features that encourage engagement

### 2. Define premium features

- Advanced functionality
- Enhanced user experience
- Exclusive content or capabilities

## Subscription Configuration Steps

Decide how many access tiers to offer based on your app's functionality and monetization goal.

### 1. Entitlements

> For more information on how to set this up, please refer to our [entitlement setup documentation](https://www.revenuecat.com/docs/getting-started/entitlements).

Reviewer: I might add a very short blurb about what entitlements are and why they're needed. I know you link to the docs, so it might not be necessary, but it will avoid a click if they're not sure.

![Screenshot](/images/playbooks/freemium/freemium-entitlements.png)

### 2. Products

> For more information on how to set this up, please refer to our [product setup documentation](https://www.revenuecat.com/docs/offerings/products-overview).

Reviewer: I might add a very short blurb about what products are and why they're needed. I know you link to the docs, so it might not be necessary, but it will avoid a click if they're not sure.

![Screenshot](/images/playbooks/freemium/freemium-products.png)

### 3. Offering

> For more information on how to set this up, please refer to our [offering setup documentation](https://www.revenuecat.com/docs/offerings/overview).

Reviewer: I might add a very short blurb about what an offering is and why it's needed. I know you link to the docs, so it might not be necessary, but it will avoid a click if they're not sure.

import flutter from "@site/code_blocks/tools/paywalls_flutter_1.dart?raw";
import kmp from "@site/code_blocks/tools/paywalls_kmp_1.kts?raw";

<RCCodeBlock
  tabs={[
    {
      type: "swift",
      title: "SwiftUI",
      content: swiftui,
    },
    {
      type: "kotlin",
      title: "Kotlin",
      content: kotlin,
    },
    {
      type: "rn",
      title: "React Native",
      content: react,
    },
    {
      type: "flutter",
      title: "Flutter",
      content: flutter,
    },
    {
      type: "kotlin",
      title: "Kotlin Multiplatform",
      content: kmp,
    },
  ]}
/>

## Best Practices

1. **Free Tier Value**

   - Provide genuine value in the free tier
   - Make core functionality accessible

2. **Premium Feature Selection**

   - Choose premium features that offer clear additional value
   - Consider different types of premium value:
     - Enhanced functionality
     - Removed limitations
     - Exclusive content
     - Better user experience

3. **User Experience**
   - Clear distinction between free and premium features
   - Smooth upgrade flow
   - Easy access to subscription management

## Test your Paywall

> - Make sure your implementation works smoothly by testing across different user scenarios:

Reviewer: ```suggestion
Ensure your implementation works smoothly by testing across different user scenarios:
```

- Free features remain accessible at all times
- A paywall is shown when a premium feature is selected
- Premium features unlock immediately after purchase
- Restore purchases and error flows work as expected

Check on our [Testing Guide](https://www.revenuecat.com/docs/guides/testing-guide/use-cases) to fine-tune your test cases and make sure everything is working as expected.

## Next Steps

> 1. Use [Experiments](https://www.revenuecat.com/docs/tools/experiments-v1) to try out different Offerings and Pricing Strategies.

Reviewer: Possibly mention Targeting / feature gating here as well.

2. Integrate our [Customer Center](https://www.revenuecat.com/docs/tools/customer-center) feature to help users manage their subscriptions.

> 3. Once you're ready to launch, take a look into this [App Subscription Launch Checklist](https://www.revenuecat.com/docs/test-and-launch/launch-checklist)

Reviewer: ```suggestion
3. Once you're ready to launch, review this [App Subscription Launch Checklist](https://www.revenuecat.com/docs/test-and-launch/launch-checklist).
```
